src/panels/lovelace/editor/hui-dialog-edit-card.ts

```diff
@@ -53,58 +72,131 @@ export class HuiDialogEditCard extends LitElement {
-          ></hui-yaml-editor>
+          ${
+            this._editorToggle && this._configElement !== null
+              ? html`<div class="element-editor">${when(
+                  this._configElement,
+                  () => this._configElement,
+                  () => html`Loading...`
+                )}</div>`
+              : html`
+              <hui-yaml-editor
+                .yaml="${this._currentConfigYaml}"
+                @yaml-changed="${this._handleYamlChanged}"
+              ></hui-yaml-editor>`
+          }
           <hui-yaml-card-preview
             .hass="${this.hass}"
-            .yaml="${this._cardConfig}"
+            .yaml="${this._currentConfigYaml}"
           ></hui-yaml-card-preview>
         </paper-dialog-scrollable>
         <div class="paper-dialog-buttons">
-          <paper-button @click="${this._closeDialog}">Cancel</paper-button>
-          <paper-button @click="${this._updateConfig}">Save</paper-button>
+          <paper-button
+            @click="${this._toggleEditor}"
+          >Toggle Editor</paper-button>
+          <paper-button
+            @click="${this._closeDialog}"
+          >Cancel</paper-button>
+          <paper-button
+            @click="${this._updateConfig}"'
+          >Save</paper-button>
         </div>
       </paper-dialog>
     `;
   }
 
-  private _handleYamlChanged(ev) {
-    this._previewEl.yaml = ev.detail.yaml;
+  private _handleYamlChanged(ev: YamlChangedEvent): void {
+    this._configValue = { format: "yaml", value: ev.detail.yaml };
+    this._updatePreview(this._configValue);
+  }
+
+  private _handleConfigChanged(value: LovelaceConfig): void {
+    if (this._configElement) {
+      this._configElement.setConfig(value);
+    }
+    this._configValue = { format: "js", value };
+    this._updatePreview(this._configValue);
+  }
+
+  private _updatePreview(value: ConfigValue) {
+    if (!this._previewEl) {
+      return;
+    }
+    this._previewEl.value = value;
   }
 
-  private _closeDialog() {
+  private _closeDialog(): void {
     this._dialog.close();
   }
 
-  private async _loadConfig() {
-    this._cardConfig = await getCardConfig(this.hass!, this._cardId!);
-    await this.updateComplete;
-    // This will center the dialog with the updated config
+  private _toggleEditor(): void {
+    if (this._editorToggle && this._configValue!.format === "js") {
+      this._currentConfigYaml = yaml.safeDump(this._configValue!.value);
+    } else if (this._configElement && this._configValue!.format === "yaml") {
+      this._configElement.setConfig(yaml.safeLoad(this._configValue!.value));
+    }
+    this._editorToggle = !this._editorToggle;
+  }
+
+  private async _loadConfig(): Promise<void> {
+    this._currentConfigYaml = await getCardConfig(this.hass!, this._cardId!);
+  }
+
+  private async _loadConfigElement(): Promise<void> {
+    const conf = yaml.safeLoad(this._currentConfigYaml);
+
+    const tag = conf.type.startsWith(CUSTOM_TYPE_PREFIX)
```

Comment:
I would have expected this code to be extracted from `create-card-element` into a helper function that both this card and `create-card-element` would depend on. That way we are guaranteed that it always stays in sync.